### PR TITLE
MEN-1289: Fixed format check to conform to the expected artifact-file…

### PR DESCRIPTION
**Pull request by oleorhagen**
Changelog: Title

Also added tests for the regexp added to executor

Signed-off-by: Ole Petter <ole.orhagen@cfengine.com>

### CODE REVIEW

**Comment by oleorhagen** [comment]
@GregorioDiStefano yo

**Comment by oleorhagen** [comment]
@kacf

File: statescript/executor.go [hunk 1]
```diff
@@ -71,7 +72,14 @@ func (l Launcher) get(state, action string) ([]os.FileInfo, string, error) {
 
 		if strings.Contains(file.Name(), state) &&
 			strings.Contains(file.Name(), action) {
-			scripts = append(scripts, file)
+
+			// all scripts must be formated like `ArtifactInstall(_Enter_05_wifi-driver)(optional)`
+			re := regexp.MustCompile(`([A-Za-z]+)_(Enter|Leave|Error)(_[0-9][0-9])?(_\S+)?`)
```
Comment (kacf) [inline statescript/executor.go]: The two digits are not optional, only the underscore and string following them.

File: statescript/executor.go [hunk 2]
```diff
@@ -71,7 +72,14 @@ func (l Launcher) get(state, action string) ([]os.FileInfo, string, error) {
...
+			if len(file.Name()) == len(re.FindString(file.Name())) {
+				scripts = append(scripts, file)
+			} else {
+				log.Warning("There was format mismatch between %s and the expected script format, so the script will not be run.", file.Name())
```
Comment (kacf) [inline statescript/executor.go]: @pasinskim: Do you agree that warning is good enough? Or do you think we should error here?

File: statescript/executor.go (hunk 1 above)
Comment (oleorhagen) [inline statescript/executor.go]: Really? Then this needs to be handled, either somewhere else, or in some other way, as the get function will receive scripts like (at least according to TestExecutor) 'Download_Enter' and 'ArtifactInstall_Leave'
```go
	// create some content in scripts directory
	_, err = createArtifactTestScript(tmpArt, "ArtifactInstall_Leave", "#!/bin/bash \ntrue")

	tmpRootfs, err := ioutil.TempDir("", "rootfs_scripts")
	assert.NoError(t, err)
	defer os.RemoveAll(tmpRootfs)

	// create some content in scripts directory
	rootfsF, err := os.Create(filepath.Join(tmpRootfs, "Download_Enter"))

```

Comment (kacf) [inline statescript/executor.go]: Hmm, that doesn't sound right. `Download_Enter` by itself is not a valid script. Maybe you can comment, @pasinskim?

Comment (pasinskim) [inline statescript/executor.go]: Yup. The digits after the name are not optional. I've been not checking the name syntax here (only while creating the artifact) so probably tests need some update as well to follow the naming convention.

File: statescript/executor.go (hunk 2 above)
Comment (pasinskim) [inline statescript/executor.go]: I think this should be enough. In fact if you are using mender-artifact tool there is no way you will end up having broken names at this point. So this is kind of "additional" safety check. BTW, the line with the warning message seems to be quite long here...

File: statescript/executor.go [hunk 3]
```diff
@@ -73,12 +73,12 @@ func (l Launcher) get(state, action string) ([]os.FileInfo, string, error) {
 		if strings.Contains(file.Name(), state) &&
 			strings.Contains(file.Name(), action) {
 
-			// all scripts must be formated like `ArtifactInstall(_Enter_05_wifi-driver)(optional)`
-			re := regexp.MustCompile(`([A-Za-z]+)_(Enter|Leave|Error)(_[0-9][0-9])?(_\S+)?`)
+			// all scripts must be formated like `ArtifactInstall_Enter_05(_wifi-driver)`(optional)
+			re := regexp.MustCompile(`([A-Za-z]+)_(Enter|Leave|Error)_[0-9][0-9](_\S+)?`)
```
Comment (oleorhagen) [inline statescript/executor.go]: Numbering the scripts is no longer optional

File: statescript/executor.go [hunk 4]
```diff
@@ -73,12 +73,12 @@ func (l Launcher) get(state, action string) ([]os.FileInfo, string, error) {
...
 			if len(file.Name()) == len(re.FindString(file.Name())) {
 				scripts = append(scripts, file)
 			} else {
-				log.Warning("There was format mismatch between %s and the expected script format, so the script will not be run.", file.Name())
+				log.Warningf("script format mismatch: '%s' will not be run ", file.Name())
```
Comment (oleorhagen) [inline statescript/executor.go]: Shortened the error message to your liking?

**Comment by kacf** [comment]
There is a conflict now, but I anyway need to test @pasinskim's branch this week, so I can take it from here.

**Comment by kacf** [comment]
Merged into https://github.com/mendersoftware/mender/pull/228, we continue from there.

@oleorhagen: Can you close this PR? This repository belongs to Marcin, so I can't close it.

**Pull request closed without merging**